support-frontend/app/utils/CheckoutValidationRules.scala

```diff
@@ -94,21 +94,25 @@ object DigitalPackValidation {
     import createSupportWorkersRequest.product._
     import createSupportWorkersRequest.billingAddress._
 
-    val allowCorporateSubs = false
-    def isCorporateSub(redemptionData: RedemptionData) =
-      allowCorporateSubs //TODO: validate redemption data, this is coming in part 2
+    val allowCorporateSubs = true
 
-    def hasValidPaymentDetails(request: CreateSupportWorkersRequest) = paymentFields.fold(
-      paymentFields => PaidProductValidation.noEmptyPaymentFields(paymentFields),
-      redemptionData => isCorporateSub(redemptionData)
-    )
+    def isValidCorporateSub(redemptionData: RedemptionData) =
+      SimpleCheckoutFormValidation.passes(createSupportWorkersRequest) &&
+        hasStateIfRequired(country, state) &&
+        allowCorporateSubs //TODO: validate redemption data here? This would mean returning a Future
```

Comment:
This is similar to promo codes and we don't validate them until CreateZuoraSubscription (in fact we didn't even validate in CreateZuoraSubscription until the one for one incident!).  This validation is to avoid support workers failures caused by invalid submissions/hacker http scanners/etc I think.

Reminds me though, someone on chat the other day was saying "parse, don't validate" as a principle (in the most general sense), and it made me think about this code.
Might be a bit much for a refactor, but food for thought.